enterprise/views.py

```diff
@@ -538,23 +538,20 @@ def get_final_price(self, mode, request):
         """
         Get course mode's SKU discounted price after applying any entitlement available for this user.
         """
-        price = mode['min_price']
-        username = request.user.username
         try:
             client = ecommerce_api_client(request.user)
-            endpoint = getattr(
-                client.enterprise.user,
-                '{username}/products/{sku}/price'.format(sku=mode['sku'], username=username)
-            )
-            price_details = endpoint.get()
-            price = price_details.get('final_price', price)
+            endpoint = client.baskets.calculate
+            price_details = endpoint.get(sku=[mode['sku']])
+            price = price_details['total_incl_tax']
+            if price != mode['min_price']:
+                return '${}'.format(price)
         except HttpClientError:
             logger.error(
                 "Failed to get price details for course mode's SKU '{sku}' for username '{username}'".format(
-                    sku=mode['sku'], username=username
+                    sku=mode['sku'], username=request.user.username
                 )
             )
-        return '${price}'.format(price=price)
+        return mode['original_price']
```

Comment:
@replaceafill The previous version of the code was returning `min_price` instead of `original_price`. Could you comment on the differences between `min_price` and `original_price`, and why we'd want to return `original_price` here (instead of `min_price`)?